### PR TITLE
ligo.skymap v2.0.0

### PR DESCRIPTION
**Pull request by duncanmmacleod**
Update to ligo.skymap 2.0.0.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by duncanmmacleod** [comment]
@lpsinger, I think this is basically ready. The linux builds all seem to work and use numpy 2.0.0rc1 for the build and then numpy 1.x for the testing, which is what we want.

However, there seems to be an unrelated (to numpy versions) issue with the macOS builds that I encourage you to take a look at:

```text
  building 'ligo.skymap.core' extension
  creating build/temp.macosx-10.9-x86_64-cpython-39
  creating build/temp.macosx-10.9-x86_64-cpython-39/src
  x86_64-apple-darwin13.4.0-clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include -fPIC -O2 -isystem /Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fstack-protector-strong -O2 -pipe -isystem /Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/work=/usr/local/src/conda/ligo.skymap-2.0.0 -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env=/usr/local/src/conda-prefix -O3 -flto -D_FORTIFY_SOURCE=2 -isystem /Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include -mmacosx-version-min=10.9 -O3 -DGSL_RANGE_CHECK_OFF -DHAVE_INLINE -DPy_LIMITED_API=50921472 -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -DNPY_NO_DEPRECATED_API=NPY_1_23_API_VERSION -I/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include -I/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/lib/python3.9/site-packages/numpy/_core/include -I/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include/python3.9 -c src/bayestar_distance.c -o build/temp.macosx-10.9-x86_64-cpython-39/src/bayestar_distance.o -std=gnu11 -fvisibility=hidden -fopenmp
  x86_64-apple-darwin13.4.0-clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include -fPIC -O2 -isystem /Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fstack-protector-strong -O2 -pipe -isystem /Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/work=/usr/local/src/conda/ligo.skymap-2.0.0 -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env=/usr/local/src/conda-prefix -O3 -flto -D_FORTIFY_SOURCE=2 -isystem /Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include -mmacosx-version-min=10.9 -O3 -DGSL_RANGE_CHECK_OFF -DHAVE_INLINE -DPy_LIMITED_API=50921472 -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -DNPY_NO_DEPRECATED_API=NPY_1_23_API_VERSION -I/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include -I/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/lib/python3.9/site-packages/numpy/_core/include -I/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include/python3.9 -c src/bayestar_moc.c -o build/temp.macosx-10.9-x86_64-cpython-39/src/bayestar_moc.o -std=gnu11 -fvisibility=hidden -fopenmp
  x86_64-apple-darwin13.4.0-clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include -fPIC -O2 -isystem /Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fstack-protector-strong -O2 -pipe -isystem /Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/work=/usr/local/src/conda/ligo.skymap-2.0.0 -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env=/usr/local/src/conda-prefix -O3 -flto -D_FORTIFY_SOURCE=2 -isystem /Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include -mmacosx-version-min=10.9 -O3 -DGSL_RANGE_CHECK_OFF -DHAVE_INLINE -DPy_LIMITED_API=50921472 -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -DNPY_NO_DEPRECATED_API=NPY_1_23_API_VERSION -I/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include -I/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/lib/python3.9/site-packages/numpy/_core/include -I/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include/python3.9 -c src/bayestar_sky_map.c -o build/temp.macosx-10.9-x86_64-cpython-39/src/bayestar_sky_map.o -std=gnu11 -fvisibility=hidden -fopenmp
  x86_64-apple-darwin13.4.0-clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include -fPIC -O2 -isystem /Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fstack-protector-strong -O2 -pipe -isystem /Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/work=/usr/local/src/conda/ligo.skymap-2.0.0 -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env=/usr/local/src/conda-prefix -O3 -flto -D_FORTIFY_SOURCE=2 -isystem /Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include -mmacosx-version-min=10.9 -O3 -DGSL_RANGE_CHECK_OFF -DHAVE_INLINE -DPy_LIMITED_API=50921472 -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -DNPY_NO_DEPRECATED_API=NPY_1_23_API_VERSION -I/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include -I/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/lib/python3.9/site-packages/numpy/_core/include -I/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include/python3.9 -c src/core.c -o build/temp.macosx-10.9-x86_64-cpython-39/src/core.o -std=gnu11 -fvisibility=hidden -fopenmp
  x86_64-apple-darwin13.4.0-clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include -fPIC -O2 -isystem /Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fstack-protector-strong -O2 -pipe -isystem /Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/work=/usr/local/src/conda/ligo.skymap-2.0.0 -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env=/usr/local/src/conda-prefix -O3 -flto -D_FORTIFY_SOURCE=2 -isystem /Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include -mmacosx-version-min=10.9 -O3 -DGSL_RANGE_CHECK_OFF -DHAVE_INLINE -DPy_LIMITED_API=50921472 -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -DNPY_NO_DEPRECATED_API=NPY_1_23_API_VERSION -I/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include -I/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/lib/python3.9/site-packages/numpy/_core/include -I/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_h_env/include/python3.9 -c src/cubic_interp.c -o build/temp.macosx-10.9-x86_64-cpython-39/src/cubic_interp.o -std=gnu11 -fvisibility=hidden -fopenmp
  src/cubic_interp.c:139:14: error: call to undeclared library function 'aligned_alloc' with type 'void *(unsigned long, unsigned long)'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
      interp = aligned_alloc(
               ^
  src/cubic_interp.c:139:14: note: include the header <stdlib.h> or explicitly provide a declaration for 'aligned_alloc'
  1 error generated.
  error: command '/Users/runner/miniforge3/conda-bld/ligo.skymap_1713429796556/_build_env/bin/x86_64-apple-darwin13.4.0-clang' failed with exit code 1
  error: subprocess-exited-with-error
```

**Comment by lpsinger** [comment]
Odd. [`aligned_alloc` was introduced in C11](https://en.cppreference.com/w/c/memory/aligned_alloc) and [we specify `-std=gnu11`](https://git.ligo.org/lscsoft/ligo.skymap/-/blob/main/setup.py?ref_type=heads#L42). What do we know about the compiler and standard library that conda is using?

**Comment by lpsinger** [comment]
Possible answer: [`aligned_alloc` may require macOS 10.15 or newer](https://stackoverflow.com/a/76984907/167694).

**Comment by lpsinger** [comment]
I switched from `posix_memalign` to `aligned_alloc` a year ago: https://git.ligo.org/lscsoft/ligo.skymap/-/commit/e8bef45ab00e93185745265540352fd1615cdc98

File: recipe/meta.yaml [hunk 1]
```diff
@@ -56,8 +59,7 @@ requirements:
     - pkg-config
     # cross-compiling:
     - cross-python_{{ target_platform }}  # [build_platform != target_platform]
-    - numpy                               # [build_platform != target_platform and (py<39 or py>=311)]
-    - numpy 1.22                          # [build_platform != target_platform and 39<=py<311]
+    - numpy >=2.0.0a0                     # [build_platform != target_platform]
```
Comment (lpsinger) [inline recipe/meta.yaml]: Why a0 and not rc1?

**Review by lpsinger — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (duncanmmacleod) [reply recipe/meta.yaml]: It's more general, but no very good reason.

**Comment by lpsinger** [comment]
I rediffed the patches in git mbox format in [cb5ba33](https://github.com/conda-forge/ligo.skymap-feedstock/pull/71/commits/cb5ba3397be44af7ee2a98e5cb52369a4975098e); the patches are no longer applying. Do you know why?

File: recipe/meta.yaml (hunk 1 above)
Comment (lpsinger) [reply recipe/meta.yaml]: OK, according to the Numpy migration guide, this should be rc1.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -66,12 +68,11 @@ requirements:
     - libcblas
     - libgomp  # [linux]
     - llvm-openmp  # [osx]
-    - numpy  # [py<39 or py>=311]
-    - numpy 1.22  # [39<=py<311]
+    - numpy >=2.0.0a0
```
Comment (lpsinger) [inline recipe/meta.yaml]:
```suggestion
    - numpy >=2.0.0rc1
```

**Review by lpsinger — CHANGES_REQUESTED** [review]

**Comment by lpsinger** [comment]
> I rediffed the patches in git mbox format in [cb5ba33](https://github.com/conda-forge/ligo.skymap-feedstock/pull/71/commits/cb5ba3397be44af7ee2a98e5cb52369a4975098e); the patches are no longer applying. Do you know why?

Never mind. I fixed it.

File: recipe/conda_build_config.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# need macOS 10.15 for aligned_alloc
+c_stdlib_version:  # [osx and x86_64]
+  - 10.15          # [osx and x86_64]
```
Comment (duncanmmacleod) [inline recipe/conda_build_config.yaml]: @lpsinger, if you're patching out the need for macOS 10.15, please remove this change and regenerate the feedstock.

**Comment by lpsinger** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml (hunk 1 above)
Comment (lpsinger) [reply recipe/meta.yaml]:
```suggestion
    - numpy >=2.0.0rc1                    # [build_platform != target_platform]
```

**Comment by lpsinger** [comment]
I think it's working now. I can't request a review from you since you are the author of the PR, but would you please look over it?

**Comment by duncanmmacleod** [comment]
> I think it's working now. I can't request a review from you since you are the author of the PR, but would you please look over it?

Looks fine, have added the `automerge` label.

**Pull request merged**